service/k8sclient/k8client.go

```diff
@@ -0,0 +1,278 @@
+// dbaas-controller
+// Copyright (C) 2020 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package k8sclient
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/AlekSi/pointer"
+	_ "github.com/percona/percona-server-mongodb-operator/pkg/apis/psmdb/v1" // It'll be implemented later.
+	pxc "github.com/percona/percona-xtradb-cluster-operator/pkg/apis/pxc/v1"
+	"github.com/pkg/errors"
+	core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/percona-platform/dbaas-controller/logger"
+	kubectl2 "github.com/percona-platform/dbaas-controller/service/k8sclient/kubectl"
+)
+
+const backupImage = "percona/percona-xtradb-cluster-operator:1.4.0-pxc8.0-backup"
+const pxcImage = "percona/percona-xtradb-cluster-operator:1.4.0-pxc8.0"
+const backupStorageName = "test-backup-storage"
+
+const perconaXtradbClusterKind = "PerconaXtraDBCluster"
+const perconaServerMongoDBKind = "PerconaServerMongoDB"
+
+// CreateXtraDBParams contains all parameters required to create percona xtradb cluster.
+type CreateXtraDBParams struct {
+	Name string
+	Size int32
+}
+
+// UpdateXtraDBParams contains all parameters required to update percona xtradb cluster.
+type UpdateXtraDBParams struct {
+	Name string
+	Size int32
+}
+
+// DeleteParams contains all parameters required to delete cluster.
+type DeleteParams struct {
+	Name string
+}
+
+// Cluster contains information related to cluster.
+type Cluster struct {
+	Name   string
+	Kind   string
```

Comment:
Let's use a special type instead of `string`

Reply:
https://github.com/percona-platform/dbaas-controller/pull/21/files#diff-0ebedeb0a1a8df09584e80193b8fcd99R65